contrib/babelfishpg_common/src/varchar.c

```diff
@@ -888,11 +888,12 @@ Datum
 varchar2time(PG_FUNCTION_ARGS)
 {
 	VarChar    *source = PG_GETARG_VARCHAR_PP(0);
+	int32		typmod = PG_GETARG_INT32(1);
```

Comment:
Instead of directly assigning typmod with the argument, we should first check if there are actually 2 arguments passed, else it will again result in Junk values.
```suggestion
	int32		typmod = -1;
	if (PG_NARGS() > 1)
		typmod = PG_GETARG_INT32(1);
```

Reply:
Instead of above suggestion, lets directly use `DEFAULT` in SQL definition of function itself.

Reply:
I think the above suggestion is backward compatible as it ensures that dumps with the older version (having only one argument) won't break after an upgrade.[reference](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/f33f75f7d07327660d39492745a384db95893993/contrib/babelfishpg_common/src/varbinary.c#L784)

Reply:
nack, just adding `DEFAULT` in SQL definition won't break any upgrades. Also, Cast only gets resolved during runtime so there will not be any issue with  dumps of older version.

@Yvinayak07 Please add upgrade  dependency test so validate the same.

Reply:
Added default value and upgrade dependency tests